testing/events.go

```diff
@@ -174,6 +174,19 @@ func ParseProposalIDFromEvents(events []abci.Event) (uint64, error) {
 	return 0, fmt.Errorf("proposalID event attribute not found")
 }
 
+// ParsePacketSequenceFromEvents parses events emitted from MsgRecvPacket and returns the packet sequence
+func ParsePacketSequenceFromEvents(events []abci.Event) (uint64, error) {
+	for _, event := range events {
+		for _, attribute := range event.Attributes {
+			if attribute.Key == "packet_sequence" {
+				return strconv.ParseUint(attribute.Value, 10, 64)
+			}
```

Comment:
True. To be honest I copied the function above which is very similar, but I could change to this. I'm not a fan of nested loops, but I am not sure if this reads a lot better, though?

Reply:
I tend to prefer structures like this but this is personal preference. If it is used more than once (I see also at least 3 more usages at the top of the file) we could extract it to an helper fn:

```go
func attributeByKey(key string, attributes []abci.EventAttribute) (Attribute, bool) {
    if idx :=  slices.IndexFunc(event.Attributes,func(a abci.EventAttribute) bool {return a.Key == key}); idx == -1 {
             return abci.Attribute{}, false
     }
     return attributes[idx], true
}
```

So that we can later just do:

```go
for _, event := range events {
		if attribute, found := attributeByKey("packet_sequence", event.Attributes); found {
				return strconv.ParseUint(attribute.Value, 10, 64)
			}
	}

```


Which I think makes things easier to read.
(now that I think about it, even if we do not use the slices.IndexFunc method, I think we could still wrap the loop over event.Attributes in an helper function)

Reply:
That does make sense! I'll look into that :)

Reply:
I'd be interested in a generic fn for parsing event attributes so we don't need to add a helper fn for each attribute, but maybe lets take this discussion to a separate issue?

Reply:
#6651